src/render/mod.rs

```diff
@@ -3551,6 +3584,187 @@ impl Draw<AlphaMask3d> for DrawEffects {
+    entries.push(BindGroupLayoutEntry {
+        binding: 0,
+        visibility: ShaderStages::COMPUTE,
+        ty: BindingType::Buffer {
+            ty: BufferBindingType::Storage { read_only: false },
+            has_dynamic_offset: false,
+            min_binding_size: Some(particle_layout_min_binding_size),
+        },
+        count: None,
+    });
+    // (1,1) IndirectBuffer
+    entries.push(BindGroupLayoutEntry {
+        binding: 1,
+        visibility: ShaderStages::COMPUTE,
+        ty: BindingType::Buffer {
+            ty: BufferBindingType::Storage { read_only: false },
+            has_dynamic_offset: false,
+            min_binding_size: BufferSize::new(12),
+        },
+        count: None,
+    });
+    // (1,2) array<ParticleGroup>
+    let particle_group_size =
+        GpuParticleGroup::aligned_size(render_device.limits().min_storage_buffer_offset_alignment);
+    entries.push(BindGroupLayoutEntry {
+        binding: 2,
+        visibility: ShaderStages::COMPUTE,
+        ty: BindingType::Buffer {
+            ty: BufferBindingType::Storage { read_only: true },
+            has_dynamic_offset: false,
+            min_binding_size: Some(particle_group_size),
+        },
+        count: None,
+    });
+    if let Some(min_binding_size) = property_layout_min_binding_size {
+        // (1,3) Properties
+        entries.push(BindGroupLayoutEntry {
+            binding: 3,
+            visibility: ShaderStages::COMPUTE,
+            ty: BindingType::Buffer {
+                ty: BufferBindingType::Storage { read_only: true },
+                has_dynamic_offset: false, // TODO
+                min_binding_size: Some(min_binding_size),
+            },
+            count: None,
+        });
+    }
+
+    trace!(
+        "Creating particle bind group layout '{}' for init pass with {} entries.",
+        label,
+        entries.len()
+    );
+    render_device.create_bind_group_layout(label, &entries)
+}
+
+fn create_init_render_indirect_bind_group_layout(
+    render_device: &RenderDevice,
+    label: &str,
+    clone: bool,
+) -> BindGroupLayout {
+    let storage_alignment = render_device.limits().min_storage_buffer_offset_alignment;
+    let render_effect_indirect_size = GpuRenderEffectMetadata::aligned_size(storage_alignment);
+    let render_group_indirect_size = GpuRenderGroupIndirect::aligned_size(storage_alignment);
+
+    let mut entries = vec![
+        // @binding(0) var<storage, read_write> render_effect_indirect :
+        // RenderEffectMetadata
+        BindGroupLayoutEntry {
+            binding: 0,
+            visibility: ShaderStages::COMPUTE,
+            ty: BindingType::Buffer {
+                ty: BufferBindingType::Storage { read_only: false },
+                has_dynamic_offset: true,
+                min_binding_size: Some(render_effect_indirect_size),
+            },
+            count: None,
+        },
+        // @binding(1) var<storage, read_write> render_group_indirect : RenderGroupIndirect
```

Comment:
```suggestion
        // @binding(1) var<storage, read_write> dest_render_group_indirect : RenderGroupIndirect
```

(this makes it easier to find the Rust code associated with the shader code if we keep the names consistent)